dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/MergeableClusterInvoker.java

```diff
@@ -114,7 +114,7 @@ protected Result doInvoke(Invocation invocation, List<Invoker<T>> invokers, Load
         if (resultList.isEmpty()) {
             return AsyncRpcResult.newDefaultAsyncResult(invocation);
         } else if (resultList.size() == 1) {
-            return resultList.iterator().next();
+            return AsyncRpcResult.newDefaultAsyncResult(resultList.iterator().next().getValue(), invocation);
```

Comment:
It's unnecessary to get element by iterator, use index to get element directly will be better.

Reply:
Yes, I cannot agree more!